bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/TableOfContentImpl.java

```diff
@@ -0,0 +1,131 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2017 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.internal.models.v1;
+
+import javax.annotation.Nonnull;
+
+import com.adobe.cq.wcm.core.components.models.TableOfContent;
+import com.adobe.cq.wcm.core.components.models.TableOfContentItem;
+import com.day.cq.wcm.api.Page;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ValueMap;
+import org.apache.sling.models.annotations.Default;
+import org.apache.sling.models.annotations.Exporter;
+import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.injectorspecific.ScriptVariable;
+import org.apache.sling.models.annotations.injectorspecific.Self;
+import org.apache.sling.models.annotations.injectorspecific.ValueMapValue;
+
+import com.adobe.cq.export.json.ComponentExporter;
+import com.adobe.cq.export.json.ExporterConstants;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Model(adaptables = SlingHttpServletRequest.class,
+    adapters = {TableOfContent.class, ComponentExporter.class},
+    resourceType = {TableOfContentImpl.RESOURCE_TYPE}
+    )
+@Exporter(name = ExporterConstants.SLING_MODEL_EXPORTER_NAME,
+    extensions = ExporterConstants.SLING_MODEL_EXTENSION
+)
+public class TableOfContentImpl implements  TableOfContent {
+
+
+    public static final String RESOURCE_TYPE = "core/wcm/components/tableOfContent/v1/tableOfContent";
+    private static final String PROP_DEFAULT_TITLE = "Table of Content";
+    private static final String TITLE_V1 = "core/wcm/components/title/v1/title";
+    private static final String TITLE_V2 = "core/wcm/components/title/v2/title";
+
+    @Self
+    private SlingHttpServletRequest slingRequest;
+
+    private List<TableOfContentItem> items;
+
+    @ValueMapValue
+    @Default(values = PROP_DEFAULT_TITLE)
```

Comment:
Default title will need to be internationalised.

Reply:
Also, currently, a custom title is not being picked up as you're referencing the wrong JCR property. See https://github.com/adobe/aem-core-wcm-components/blob/master/bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/TitleImpl.java#L69